src/main/java/com/adyen/httpclient/HttpURLConnectionClient.java

```diff
@@ -269,7 +269,7 @@ private void installCertificateVerifier(URLConnection connection, InputStream te
             try {
                 // Create new KeyStore for the terminal certificate
                 CertificateFactory certificateFactory = CertificateFactory.getInstance("X.509");
-                terminalCertificateStream.reset();
```

Comment:
Was there actually a scenario for the `reset`, I couldn't find a `mark`?
If yes, we could use something like `if terminalCertificateStream.markSupported()`.

Reply:
I think it was just done considering that we will always have stream from beginning, but not all streams support reset. Also even if mark is supported but if it is not set, it will throw exception on reset.